pkg/providers/instance/offerings/commonerrorhandlers.go

```diff
@@ -14,102 +14,67 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package instance
+package offerings
 
 import (
 	"context"
 	"fmt"
 	"strings"
 	"time"
 
-	sdkerrors "github.com/Azure/azure-sdk-for-go-extensions/pkg/errors"
+	"github.com/Azure/karpenter-provider-azure/pkg/cache"
 	"github.com/Azure/skewer"
 	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
 	corecloudprovider "sigs.k8s.io/karpenter/pkg/cloudprovider"
 )
 
-type responseErrorHandler struct {
-	name                string
-	matchError          func(error) bool
-	handleResponseError func(ctx context.Context, provider *DefaultVMProvider, sku *skewer.SKU, instanceType *corecloudprovider.InstanceType, zone, capacityType string, responseError error) error
-}
+var (
+	SubscriptionQuotaReachedReason              = "SubscriptionQuotaReached"
+	AllocationFailureReason                     = "AllocationFailure"
+	ZonalAllocationFailureReason                = "ZonalAllocationFailure"
+	OverconstrainedZonalAllocationFailureReason = "OverconstrainedZonalAllocationFailure"
+	OverconstrainedAllocationFailureReason      = "OverconstrainedAllocationFailure"
+	SKUNotAvailableReason                       = "SKUNotAvailable"
+
+	SubscriptionQuotaReachedTTL = 1 * time.Hour
+	AllocationFailureTTL        = 1 * time.Hour
+	SKUNotAvailableSpotTTL      = 1 * time.Hour
+	SKUNotAvailableOnDemandTTL  = 23 * time.Hour
+)
+
+type commonErrorHandle func(ctx context.Context, unavailableOfferings *cache.UnavailableOfferings, sku *skewer.SKU, instanceType *corecloudprovider.InstanceType, zone, capacityType, errorCode, errorMessage string) error
```

Comment:
minor:

```suggestion
type errorHandler func(ctx context.Context, unavailableOfferings *cache.UnavailableOfferings, sku *skewer.SKU, instanceType *corecloudprovider.InstanceType, zone, capacityType, errorCode, errorMessage string) error
```

maybe?

Reply:
IMO it's OK to have 2 different handlers (with different names, obviously). With that said if you feel strongly: `errorHandlerFunc` is IMO better than `errorHandle`. My ranking is:
`errorHandler`, `errorHandlerFunc`, ..., `errorHandle`.